src/Aquarius.Client.UnitTests/Samples/Client/LazyGetTests.cs

```diff
@@ -0,0 +1,226 @@
+        {
+            if (resultCount >= totalCount)
+                throw new ArgumentOutOfRangeException(nameof(resultCount), $"ResultCount={resultCount} must be strictly less than TotalCount={totalCount}");
+
+            return CreateResults(resultCount, totalCount);
+        }
+
+        private ThingResults CreateResultsWithNoCursor(int resultCount, int totalCount)
+        {
+            var results = CreatePartialResults(resultCount, totalCount);
+
+            results.Cursor = null;
+
+            return results;
+        }
+
+        private ThingResults CreateResultsWithEmptyCursor(int resultCount, int totalCount)
+        {
+            var results = CreatePartialResults(resultCount, totalCount);
+
+            results.Cursor = string.Empty;
+
+            return results;
+        }
+
+        private ThingResults CreateResults(int resultCount, int totalCount)
+        {
+            var results = new ThingResults
+            {
+                Cursor = _fixture.Create<string>(),
+                DomainObjects = new List<Thing>(),
+                TotalCount = totalCount
+            };
+
+            if (resultCount > 0)
+            {
+                results.DomainObjects = _fixture.CreateMany<Thing>(resultCount).ToList();
+            }
+            else
+            {
+                results.Cursor = null;
+            }
+
+            return results;
+        }
+
+        private List<Thing> EvaluateAllLazyLoadedItems()
+        {
+            return _client.LazyGet<Thing, GetThings, ThingResults>(new GetThings()).DomainObjects.ToList();
+        }
+
+        private void AssertExpectedItemsAndLazyFetches(List<Thing> items, int expectedTotalCount, int expectedFetchesCount)
+        {
+            items.Count.ShouldBeEquivalentTo(expectedTotalCount, "Total count of items should match");
+
+            _mockServiceClient
+                .Received(expectedFetchesCount)
+                .Get(Arg.Any<GetThings>());
+        }
+
+        [Test]
+        public void LazyGet_SecondResponseFails_Throws()
+        {
+            _mockServiceClient
+                .Get(Arg.Any<GetThings>())
+                .Returns(
+                    x => CreatePartialResults(1, 2),
+                    x => { throw new ArgumentException();});
+
+            ((Action) (() => EvaluateAllLazyLoadedItems())).ShouldThrow<ArgumentException>();
+        }
+
+        [Test]
+        public void LazyGet_SecondResponseContainsAllRemainingItems_Succeeds()
+        {
+            _mockServiceClient
+                .Get(Arg.Any<GetThings>())
+                .Returns(
+                    x => CreatePartialResults(1, 3),
+                    x => CreateFinalResults(1, 2));
+
+            var items = EvaluateAllLazyLoadedItems();
+
+            AssertExpectedItemsAndLazyFetches(items, 2, 2);
+        }
+
+        [Test]
+        public void LazyGet_SecondResponseWithNoCursor_Succeeds()
```

Comment:
I hadn't looked at `LazyGet` because it wasn't in the PR, as I was using the LazyReview algorithm. I don't think you need to remove this test, but I'd suggest adding one that tests the code in a way closer to what the samples server will return.

Now that I have looked over `LazyGet`, there is one thing I noticed. It is comparing the total count to the number returned. This assumes that the total count and the number of items won't change between subsequent calls, and I'm not sure this is true. For example, a first call might report 4 total results, return 3, and a second call might report a total of 5 and return 2. It might also return none. There could be variations where results at the end of list are skipped accidentally.

I think it would be safest to remove the comparison of the total fetched items, and instead call the server until it returns either no continuation token or no results. I think you should also add a test for this case.